### DISCUSSION TITLE
Prototype Assembly support for Clangd

### DISCUSSION THREAD

**Discussion by BenjaminGrayNp1** [clangd/clangd#1805] [Show and tell]
Hi all, I recently posted about adding Assembly support to Clangd [on the forum](https://discourse.llvm.org/t/prototype-assembly-support-for-clangd/74417/1), and they suggested I also post it here for visibility. I am keen for it to eventually reach upstream, though the current implementation is just a prototype and focused on getting things working at all.

https://github.com/BenjaminGrayNp1/llvm-project

![Screenshot from 2023-10-26 14-26-11](https://github.com/clangd/clangd/assets/109320246/5e5434e2-d035-49dd-a6fe-fc421ba2f379)
![Screenshot from 2023-10-26 14-26-50](https://github.com/clangd/clangd/assets/109320246/4b1bc8c8-5a0b-403b-a5f0-2909e7c351b5)
![Screenshot from 2023-10-26 14-27-54](https://github.com/clangd/clangd/assets/109320246/f3d09ca1-9c17-457b-aec1-6f5f6303109b)

**Comment by HighCommander4** [comment]
Thanks for sharing!

I can't speak for the project, but my opinion as a user is that functionality like this should be in scope for clangd's feature set. There is precedent for C language family editors supporting assembly (e.g. Eclipse CDT does), and there is a fair amount of opportunity to reuse code between assembly support and C/C++ support (e.g. the preprocessor).

One thing that may help advance discussions about upstreaming this is a document outlining how this support works at a high level and the design decisions made in the implementation.

**Reply by BenjaminGrayNp1** [reply]
Right, I've added a description to the README

**Reply by HighCommander4** [reply]
Thanks! Linking for reference: https://github.com/BenjaminGrayNp1/llvm-project#prototype-clangd--assembly-project

**Comment by nickdesaulniers** [comment]
Cool!

**Comment by XutaxKamay** [comment]
Too bad this isn't discussed anymore.
I have a lot of projects where this could be useful.
